amida-tech/indaba-client
# [INBA-903] Changes that remove whitespace at the end of a survey and

#### What does this PR do?
Checks and prevents the creation of a new project if the survey name is already in use. Also removes trailing whitespaces in both project and survey names to prevent duplications.

#### Related JIRA tickets:
INBA-903

#### How should this be manually tested?
Go to the Create Project Wizard:

- Upon seeing the modal, try to immediately click "Save". The user should be stopped with warnings about the project name and survey title being required.
- Try entering just the project name but no survey name. The user should get a warning.
- Try entering just the survey name but no project name. The user should get a warning.
- Try entering a project name that is already used but a unique survey name. The user should get a warning about the need for a unique project name.*
- Try entering a survey name that is already used but a unique project name. The user should get a warning about the need for a unique survey name.*
- Try entering a survey AND a project name that are already used. The user should get a warning about both.
-Try entering a unique survey name and unique project name and saving. It should save successfully.
- Repeat the above using the "enter" key to submit instead of hitting the "Save" button.

For both Project Management and Create Project Wizard:
Click on the Project Title Modal (the card that also reads active/inactive). Try the following scenarios:
- Confirm that the modal opens with the current project name.
- Try entering a blank name. You will be stopped because a title is required.
- Try entering a project name already in use. You will be stopped. 
- Try entering a unique name. It will be accepted.
- Repeat the above using the "enter" key to submit instead of hitting the "Save" button.

Click on the Survey Title Modal (the card that also reads active/inactive). Try the following scenarios:
- Confirm that the modal opens with the current survey name.
- Try entering a blank name. You will be stopped because a title is required.
- Try entering a survey name already in use. You will be stopped.*
- Try entering a unique name. It will be accepted.
- Repeat the above using the "enter" key to submit instead of hitting the "Save" button.

***- You may have residual data from earlier uses, because survey service does not enforce unique survey names and you may have reused titles in the past. If you see "Survey title already used" type errors but not among your existing projects, know that this problem is fairly self-correcting. You may need to wipe everything however.**

#### Background/Context

#### Screenshots (if appropriate):
